plugins/ipam/dhcp/lease.go

```diff
@@ -344,6 +358,23 @@ func (l *DHCPLease) maintain() {
 	}
 }
 
+// checkLinkExistsWithBackoff uses backoffRetry to check if a network link exists
+func (l *DHCPLease) checkLinkExistsWithBackoff(ctx context.Context) error {
+    checkFunc := func() (*nclient4.Lease, error) {
+		// Returns the error to trigger a retry.
+        if _, err := netlink.LinkByName(l.link.Attrs().Name); err != nil {
+            return nil, err
+        }
+        return nil, nil
+    }
+
+    ctx, cancel := context.WithTimeout(ctx, linkCheckTotalTimeout)
+    defer cancel()
+    _, err := backoffRetry(ctx, linkCheckRetryMax, checkFunc)
```

Comment:
the only downside here is that `backoffRetry()` sets a "base delay" using the `resendDelay0` const which was from the DHCP RFC stuff, so, it's 4 seconds the first time. This could probably be quicker? Will happily rejig backoffRetry to take a base delay parameter.

Reply:
oops missed a major part of this... fix incoming.